exporter/signalfxexporter/translation/translator.go

```diff
@@ -332,10 +364,63 @@ func createDimensionsMap(rules []Rule) map[string]string {
 			return tr.Mapping
 		}
 	}
+	return nil
+}
 
+func processRules(rules []Rule) error {
+	for i, tr := range rules {
+		if tr.Action == ActionDropDimensions {
+			processedRule := Rule{Action: ActionDropDimensions}
+
+			// Set metric name filter, if metric name(s) are specified on the rule.
+			// When "drop_dimensions" actions is not scoped to a metric name, the
+			// specified dimensions will be globally dropped from all datapoints
+			// irrespective of metric name.
+			if metricNames := getMetricNamesAsSlice(tr.MetricName, tr.MetricNames); len(metricNames) > 0 {
+				metricMatcher, err := dpfilters.NewStringFilter(metricNames)
+				if err != nil {
+					return fmt.Errorf("failed creating metric matcher: %w", err)
+				}
+				processedRule.metricMatcher = metricMatcher
+			}
+
+			// Walk through WithoutDimensionPairs on the rule and create respective filters
+			// for dimension keys provided. If values for a particular key turns out to be
+			// empty, the dimension should be dropped irrespective of the value, in such cases
+			// set the dimensionMatcher to nil
+			dimensionMatcher := map[string]*dpfilters.StringFilter{}
+			for key, values := range tr.WithoutDimensionPairs {
+				if len(values) == 0 {
+					dimensionMatcher[key] = nil
+					continue
+				}
+				stringFilter, err := dpfilters.NewStringFilter(values)
```

Comment:
Do we need to support regexps or blobs for dimensions for our use case? If not, we could probably safe a few CPU cycles avoiding doing this. What do you think?

Reply:
We don't for our particular usecase. Updated to support only strings.